### ISSUE TITLE
Navbar Customization - Nav links for About and Guides

### ISSUE THREAD

**Issue by mheppler** [IQSS/dataverse#3637]
Based on feedback from the community (#2003), we will provide configurable settings to allow for the customization of the following features in the Dataverse application's navbar:
- ~Name: branding needs to be more flexible~ (out of scope, delivered in #3611)
- ~Logo: display a logo in the navbar in place of orange Dataverse logo~ (Moving to #3639 as this will be an external file)
- About link: configurable link to an installation's custom project site
  - By default, the About link in the navbar is turned off, installations config the URL to display it
- Guides links: configurable link to an installation's custom version of guides (as suggested in #2831, delivered as part of #3540 55fc1286c8c5152895f24f41fe642327e2233caa)
  - only User Guide in navbar, other guides will be linked from Dataverse.org

**Comment by djbrooke** [comment]
Thanks for creating this one, @mheppler. Should we close #2831?

**Comment by mheppler** [comment]
@djbrooke done

**Comment by pdurbin** [comment]
With regard to customization of links to the Guides, we should probably consider not only #2831 but also #2832 which is a bit different.

**Comment by dlmurphy** [comment]
Last month the Design team put together a document summarizing research insights that can help inform this feature:
https://docs.google.com/a/harvard.edu/document/d/17GqVnhUw8glb0ikQ5pYX07ed5LEs2M5LYZkAkuNvUnI/edit?usp=sharing

**Comment by raprasad** [comment]
The following navbar link updates are included:

- The "About" link no longer shows up by default.  To make the link available, set a url using the setting ```:NavbarAboutUrl```
- "Guides" is no longer a dropdown.  Instead there is a single link to a "User Guide."
  - As of release 4.6.1, this link is configurable via the settings:
    -  ```:GuidesBaseUrl``` 
    - ```:GuidesVersion```.  
    - For details, see:  http://guides.dataverse.org/en/4.6.1/installation/config.html?highlight=guidesversion


/cc @shlake

**Comment by mheppler** [comment]
Concerns regarding the notifications in comment above from @dlmurphy have been reviewed with the changes from @pdurbin and it looks like all User Guides links are properly pointing to the correct hosted guide and version, as set in the configurations.

There is also the open issue "Notifications: Ability to modify included text" #2710 from @shlake that looks at the notifications in more detail.

**Comment by sekmiller** [comment]
Code looks good. Passing to QA. There's still a file called 'navbar.html' that doesn't seem to be used currently. It should probably be deleted from the project at some point.

**Comment by sekmiller** [comment]
Also, branch is a couple of commits behind dev at the moment. It needs another update.

**Comment by mheppler** [comment]
@sekmiller I will update the branch now with develop.

Re:navbar.html -- I think that file you are referencing is only part of the UI for the docs, and not the app.

![screen shot 2017-05-31 at 4 22 04 pm](https://cloud.githubusercontent.com/assets/687227/26652264/5158c9e0-461d-11e7-8e7b-8fd72b5b9ece.png)

**Comment by pdurbin** [comment]
I ran the code locally as of 32a90dd and it looks good. I fixed the title of this issue since the stated scope has decreased (nothing about the logo anymore) but a note from @raprasad to @kcondon (or a conversation) would be appreciated since I noticed many changes of `settingsService` to `settingsWrapper` like this:

- settingsWrapper.getValueForKey(SettingsServiceBean.Key.ComputeBaseUrl) + getSwiftContainerName()
- settingsWrapper.isTrueForKey(SettingsServiceBean.Key.DatasetPublishPopupCustomTextOnAllVersions, false);
- signupAllowed = settingsWrapper.isTrueForKey(SettingsServiceBean.Key.AllowSignUp, safeDefaultIfKeyNotFound);
-  etc, etc. 

These are welcome performance improvements so I'll add the appropriate GitHub label.
